src/rgw/rgw_rest_s3.cc

```diff
@@ -1729,7 +1729,7 @@ void RGWListBucket_ObjStore_S3::send_versioned_response()
   s->formatter->dump_string("KeyMarker", marker.name);
   s->formatter->dump_string("VersionIdMarker", marker.instance);
   if (is_truncated && !next_marker.empty()) {
-    s->formatter->dump_string("NextKeyMarker", next_marker.name);
+    dump_objname(s ,encode_key, "NextKeyMarker", next_marker.name);
```

Comment:
these s3 list-object apis already do this conditional url-encoding for several other fields like Delimiter, Prefix, Key, etc.
for example, https://github.com/ceph/ceph/blob/7a2be8a9/src/rgw/rgw_rest_s3.cc#L1725-L1729:
```c++
      if (encode_key) {
        s->formatter->dump_string("Prefix", url_encode(pref_iter->first, false));
      } else {
        s->formatter->dump_string("Prefix", pref_iter->first);
      }
```
i don't mind that if-else form. but if we're adding a helper function that does the same thing, it would be nice to use the same function for all of these cases. that would involve choosing a more general name than `dump_objname()`, and supporting the same `bool encode_slash = true` argument that `url_encode()` does

Reply:
edited. pick `dump_urlsafe` as the function name for now.